packages/protocol/src/action/request/Invoke.ts

```diff
@@ -123,7 +168,10 @@ export function Invoke(
 export namespace Invoke {
     export interface Definition {
         /** List of commands to invoke */
-        commands: Invoke.CommandRequest<any>[];
+        commands?: Invoke.CommandRequest<any>[];
+
+        /** List of prepared raw commands to invoke */
+        invokeRequests?: InvokeCommandData[];
```

Comment:
It looks like the only place this is used is the adapter in `InteractionClient`...  That adapter appears to have everything necessary to construct a `CommandRequest` directly, it's just doing some of the work of assembling the commands itself

Since this screws up the public interface I think it makes sense to remove this and modify `InteractionClient` to use this interface as designed, which should reduce complexity without too much effort.  I'll add further detail in `InteractionClient`